internal/tiltfile/tiltfile_state.go

```diff
@@ -772,22 +822,22 @@ func newK8SObjectSelector(apiVersion string, kind string, name string, namespace
 		}
 	}
 
-	ret.apiVersion, err = regexp.Compile(makeCaseInsensitive(apiVersion))
+	ret.apiVersion, err = regexp.Compile(reAnchor(makeCaseInsensitive(apiVersion)))
```

Comment:
have you considered changing the matches() function to look for a full-string match, rather than mutate the regexp?